10-r-code.Rmd

```diff
@@ -47,6 +47,23 @@ This encourages interoperability and simplifies your own package development.
 If a new representation is needed, see the [Essential S4 interface](https://bioconductor.org/developers/how-to/efficient-code/#essential-s4-interface) section of [Robust and Efficient Code](https://bioconductor.org/developers/how-to/efficient-code).
 In general, [_Bioconductor_](https://bioconductor.org/) will insist on interoperability with [Common Classes](https://bioconductor.org/developers/how-to/commonMethodsAndClasses/) for acceptance.
 
+Developers should make an effort to re-use generics that fit the generic
+contract for the proposed class-method pair i.e., the behavior of the method
+fits the originally proposed behavior of the generic.
+`r BiocStyle::Biocpkg("BiocGenerics")` lists commonly used generics in
+Bioconductor. Re-using generics reduces the amount of new generics by
+consolidating existing operations.
```

Comment:
Maybe indicate something like 'It is always a mistake to introduce a new generic with the same name as an existing generic or function' because the new generic may mask, or be masked by, the previous definitions in hard-to-diagnose ways.